ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/recovery/arjunacore/XARecoveryModule.java

```diff
@@ -1223,6 +1221,40 @@ private void saveContactedJndiName(final String jndiName) {
             contactedJndiNames.add(jndiName);
         }
     }
+    
+    private String getXidLogInfo (Xid xid) {
+        StringBuilder stringBuilder = new StringBuilder();
+        
+        if (xid != null) {
+            byte[] globalTransactionId = xid.getGlobalTransactionId();
+            byte[] branchQualifier = xid.getBranchQualifier();
+            stringBuilder.append("< ");
+            stringBuilder.append(xid.getFormatId());
```

Comment:
If there is the check for `globalTransactionId` and `branchQualifier` in this printing, it could be good to check the format for `null` as well here.

Reply:
xid.getFormatId() is int